.github/workflows/nightly-tests.yml

```diff
@@ -74,23 +90,31 @@ jobs:
           vault-token: ${{ steps.vault-token-step.outputs.vault-token }}
           target: wsmtest
 
-      # The ArgoCD sync triggers the synchronization of the wsmtest cluster. Typically,
-      # the cluster is reset within a few seconds. However, it is possible for it to take
-      # minutes or hours. We have no way to check that the sync is complete. We sleep
-      # for a bit and just accept that in some crazy long cases, these tests will fail.
+      # The ArgoCD sync triggers the synchronization of the wsmtest cluster. The ArgoCD CLI
+      # will wait for the initial sync to complete, but it won't necessarily wait for WSM's
+      # new version to actually come fully online--which could take minutes or hours in rare
+      # cases. We sleep for two minutes to try to allow WSM to come online.
       # The sync is bracketed by /version probes so we have the pre and post sync versions
```

Comment:
Could you put a step here specifically to ping WSM somehow until it's online? For example, it could curl the unauthenticated `/status` endpoint.

Reply:
WSM doesn't go down, it's that the rolling zero-downtime eventually-consistent update across an effectively arbitrary number of replicas is hard to introspect without additional work.

We're actually working on that problem, among others, for our "on-demand Terra environments with specific app versions" project. My ticket here is just for the proxy-authorization header so I can get that work unblocked, and I figured a switch to Argo's CLI would be an easy win while I was at it. I'm on a bit of a time crunch so if you think it makes sense to do a narrower change and pass the header through the old curl command, I'm fine with that too. This flow for interacting with Argo isn't one we actively support--we just don't want to break it if we can help it.

Reply:
Good idea, but I think it is out of scope for this change.

Reply:
No, you're good. I just didn't approve b/c I wasn't a designated reviewer.